src/hooks/useCodapState.tsx

```diff
@@ -258,13 +290,40 @@ export const useCodapState = () => {
     }
   }, [selectedDataSet]);
 
+  const editCaseValue = async (newValue: string, caseObj: IProcessedCaseObj, attrTitle: string) => {
+    let request;
+
+    try {
+      request = await updateCaseById(selectedDataSetName, caseObj.id, {[attrTitle]: newValue});
+      if (request.success) {
+        runInAction(() => caseObj.values.set(attrTitle, newValue));
+      }
+    } catch (e) {
+      console.error("Case not updated: ", e);
+    }
+
+    return request;
+  };
+
+  const handleSetCollections = (colls: ICollection[]) => {
+    const newCollectionModels = colls.map((coll: ICollection) => {
+      const { areParentChildLinksConfigured, attrs, caseName, cases: _cases, childAttrName,
+              collapseChildren, guid, id, name, parent, title, type } = coll;
+      return {
+        areParentChildLinksConfigured, attrs, caseName, cases: _cases, childAttrName,
+        collapseChildren, guid, id, name, parent, title, type
+      };
+    });
+    applySnapshot(collections, newCollectionModels);
```

Comment:
This looks like the same code that is in `updateCollections`, can you consolidate it?